src/stocktwits_data_extractor.py

```diff
@@ -47,44 +51,83 @@ def main():
+        refresh_attempts = 120*12
+        while refresh_attempts:
 
-        for a in findAllInSoup:
+            pageContent = driver.page_source
+            soup = BeautifulSoup(pageContent, features="html.parser")
+            print('got soup')
 
-            sentimentDiv = a.find('div', attrs={'class': 'lib_XwnOHoV lib_3UzYkI9 lib_lPsmyQd lib_2TK8fEo'})
-            if sentimentDiv is None:
-                continue
+            #Find all tweets
+            findAllInSoup = soup.find_all(attrs={'class':'st_VgdfbpJ st_31oNG-n st_3A22T1e st_vmBJz6-'})
 
-            sentiment = sentimentDiv.get_text()
+            for a in findAllInSoup:
 
-            userDiv = a.find('a', attrs={'class':'st_x9n-9YN st_2LcBLI2 st_1vC-yaI st_1VMMH6S'})
-            user = userDiv.find('span').get_text()
+                messageIdA = a.find('a', attrs={'class': 'st_28bQfzV st_1E79qOs st_3TuKxmZ st_3Y6ESwY st_GnnuqFp st_1VMMH6S'})
+                messageId = messageIdA.attrs['href']
+                if messageId in messageChecklist:
+                    print('No new tweets')
+                    break
 
-            messageIdA = a.find('a', attrs={'class': 'st_28bQfzV st_1E79qOs st_3TuKxmZ st_3Y6ESwY st_GnnuqFp st_1VMMH6S'})
-            messageId = messageIdA.attrs['href']
-            
-            content = a.find('div', attrs={'class':'st_3SL2gug'}).get_text()
+                sentimentSpan = a.find('span', attrs={'class':'st_11GoBZI'})
+                if sentimentSpan is None:
+                    continue
+                sentimentDiv = sentimentSpan.find('div', attrs={'class': 'lib_XwnOHoV lib_3UzYkI9 lib_lPsmyQd lib_2TK8fEo'})
+                sentiment = sentimentDiv.get_text()
 
-            print(f"Sentiment: {sentiment}\nUser: {user}\nMessageId: {messageId}\nContent: {content}\n")
+                content = a.find('div', attrs={'class':'st_3SL2gug'}).get_text()
+                if len(re.findall(r'\w+', content)) < 4:
+                    continue
 
-            data.append((user, messageId, sentiment, content))
+                userDiv = a.find('a', attrs={'class':'st_x9n-9YN st_2LcBLI2 st_1vC-yaI st_1VMMH6S'})
+                user = userDiv.find('span').get_text()
+
+                dateScraped = time.strftime("%Y/%m/%d", time.localtime())
+                timeScraped = time.strftime("%H:%M:%S", time.localtime())
+
+                print(f"Sentiment: {sentiment}\nUser: {user}\nMessageId: {messageId}\nContent: {content}\nDate: {dateScraped}\nTime: {timeScraped}\n")
+
+                data.append((user, messageId, sentiment, content, dateScraped, timeScraped))
+                messageChecklist.append(messageId)
+
+            print('Going to sleep.', refresh_attempts, 'refreshes left.')
+            refresh_attempts-=1
+            time.sleep(time_between_refreshes)
+
+            try:
+                element = driver.find_element_by_xpath('/html/body/div[2]/div/div/div[3]/div[2]/div/div[1]/div[2]/div/div/div[2]/div[2]/div/div')
```

Comment:
Would it not be better to get the div element by using find() function and using class name?

![stocktwits-refresh](https://user-images.githubusercontent.com/10214130/86964911-ad7da280-c15e-11ea-83ac-65306cebbd39.png)

Reply:
In theory, yes, but when I tried this it didn't work, so I used the xpath rather than spending more time troubleshooting. Might be something to explore when not so pressed for time.

Reply:
Actually, this is more pressing than I realised - the xpath appears to be different in windowed mode to in fullscreen. Will deal with it today.

Reply:
Attempting to use class name raises a NoSuchElementException. Unsure why this is, so have left the xpath in for now, as it currently works.

Reply:
New Posts button has multiple classes, which apparently means find_element_by_class won't work with it. The common solution is apparently to use find_element_by_css_selector instead, which is what I have now done.